### PR TITLE
Inserted a delete clientsslprofile route and funcs

### PR DESCRIPTION
**Pull request by damnski**
Inserted a delete clientsslprofile route and funcs

Adjusted get/show to return a 404 if item not found

### CODE REVIEW

File: api/handlers_ltm.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ func (s *server) ShowClientSSLProfile(w http.ResponseWriter, r *http.Request) {
 		handleError(w, err)
 	}
 
+	if out == nil {
+		handleError(w, apierror.New(apierror.ErrNotFound, fmt.Sprintf("%s not found", name), nil))
+		return
```
Comment (tenyo) [inline api/handlers_ltm.go:69]: Do you need a return after `handleError`?

File: ltm/profiles.go [hunk 2]
```diff
@@ -105,6 +105,40 @@ func (l *LTM) ImportKey(name, thisYear string) error {
 	return nil
 }
 
+// RemoveKey removes Key from System SSL
+func (l *LTM) RemoveKey(name string) error {
+	if name == "" {
+		return apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	err := l.Service.DeleteKey(name)
```
Comment (tenyo) [inline ltm/profiles.go:114]: 👀 Why not do `if err := ...` here?

File: ltm/profiles.go [hunk 3]
```diff
@@ -105,6 +105,40 @@ func (l *LTM) ImportKey(name, thisYear string) error {
...
+	return nil
+}
+
+// RemoveCertificate removes Certificate from System SSL
+func (l *LTM) RemoveCertificate(name string) error {
+	if name == "" {
+		return apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	err := l.Service.DeleteCertificate(name)
```
Comment (tenyo) [inline ltm/profiles.go:131]: 👀 same

File: ltm/profiles.go [hunk 4]
```diff
@@ -158,3 +192,20 @@ func (l *LTM) CreateClientSSLProfile(ClientSSLProfileName, DefaultsFrom, Chain,
...
+
+	if err := l.Service.DeleteClientSSLProfile(ClientSSLProfileName); err != nil {
+		msg := fmt.Sprintf("failed to delete client-ssl profile %s on %s", ClientSSLProfileName, l.Host)
+		return apierror.New(apierror.ErrInternalError, msg, err)
+	}
+
+	log.Infof("deleted client-ssl profile %s on host %s\n", ClientSSLProfileName, l.Host)
+
+	return nil
+
```
Comment (tenyo) [inline ltm/profiles.go:210]: ✂️

File: api/orchestration.go [hunk 5]
```diff
@@ -110,3 +111,32 @@ func (o *ltmOrchestrator) createClientSSLProfile(ctx context.Context, data *Modi
...
+		return err
+	}
+
+	err = o.client.RemoveCertificate(clientSSLProfile.Cert)
+	if err != nil {
+		return err
+	}
+
+	err = o.client.RemoveKey(clientSSLProfile.Key)
+	if err != nil {
```
Comment (tenyo) [inline api/orchestration.go:137]: I think all these calls can be `if err = o.client.XXX` for brevity

**Review by tenyo — COMMENTED** [review]
Looks good to me, just a few minor things

File: api/handlers_ltm.go (hunk 1 above)
Comment (fishnix) [reply api/handlers_ltm.go:69]: yes, handleError doesnt return from the handler

File: api/handlers_ltm.go [hunk 6]
```diff
@@ -162,3 +167,35 @@ func (s *server) CreateClientSSLProfile(w http.ResponseWriter, r *http.Request)
...
+	orch := &ltmOrchestrator{
+		client: ltmService,
+	}
+
+	if err := orch.deleteClientSSLProfile(r.Context(), name); err != nil {
+		handleError(w, err)
+		return
+	}
+
+	out := []byte(fmt.Sprintf("deleted client-ssl profile %s on host %s", name, host))
```
Comment (fishnix) [inline api/handlers_ltm.go:196]: minor, but I would just put this inside of the `w.Write()` instead of creating the `out` variable.

**Review by fishnix — APPROVED** [review]
Just the minor stuff, otherwise looks good. 👍

File: api/handlers_ltm.go (hunk 1 above)
Comment (damnski) [reply api/handlers_ltm.go:69]: Thanks guys.

**Pull request merged**